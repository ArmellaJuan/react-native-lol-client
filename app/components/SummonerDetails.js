import React, { Component, PropTypes } from 'react';
import { StyleSheet,  Image, View, Text, Button } from 'react-native';

import RowInfo from './RowInfo';
import Util from '../util/Util.js';

export default class SummonerDetails extends Component{

  constructor(props) {
    super(props);
  
  }

  render() {

    
    let summoner = this.props.summoner;
    let statistics = summoner.statistics;

    return (
        <View  style = { styles.details }>
          <View style = { styles.headerProfile} >
            <Image style={ styles.image } source={{uri: summoner.profileIconUrl}}  />
            <Text style ={ styles.headerText } > {summoner.name} </Text>
          </View>
          <RowInfo labelStyle={ [styles.label] } label='Division Name' value={statistics.divisionName} /> 
          <RowInfo labelStyle={ [styles.label] } label='Tier' value={statistics.tier} /> 
          <RowInfo labelStyle={ [styles.label] } label='Wins' value={statistics.wins} /> 
          <RowInfo labelStyle={ [styles.label] } label='Losses' value={statistics.losses} />
          <RowInfo labelStyle={ [styles.label] } label='LP' value={statistics.leaguePoints} />
          <Button
            onPress={ this.props.onRecentGames } 
            title="Recent Games"
            color="dodgerblue"
            style={ styles.summonerDetailButton }
          />
        </View >
      
    );
  }

}

SummonerDetails.propTypes ={
  summoner: PropTypes.object.isRequired,
  onRecentGames: PropTypes.func.isRequired
};

const styles = StyleSheet.create({
  summonerDetailButton: {
    margin: 20
  },
  headerText: {
    fontSize:  14,
    color: 'dodgerblue',
    textAlign: 'center',
    padding: 10
  },
  image: {
    width: 70, 
    height: 70,
    borderRadius: 35
  },
  headerProfile: {
    alignItems: 'center',
  },
  details:{
    flexDirection: 'column',
    backgroundColor: 'white',
    padding: 20,
    elevation   : 3
  },
  label: {
    color: 'dodgerblue'
  }
});

