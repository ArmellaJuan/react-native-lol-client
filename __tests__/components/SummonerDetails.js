import 'react-native';
import React from 'react';
import SummonerDetails from '../../app/components/SummonerDetails';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';



test('renders correctly', () => {

  let summoner = {
    id: 55899,
    name: "Test",
    profileIconUrl: "http://ddragon.leagueoflegends.com/cdn/7.2.1/img/profileicon/553.png",
    statistics: {
      divisionName: "Trundle's Dervish",
      leaguePoints: 26,
      losses: 13,
      tier: "SILVER",
      wins: 21
    }
  };


  const tree = renderer.create(
    <SummonerDetails summoner = { summoner } onMatchHistory = { () => {} } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});