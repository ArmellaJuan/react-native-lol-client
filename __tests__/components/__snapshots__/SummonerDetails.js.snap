exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 3,
      "flexDirection": "column",
      "padding": 20,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }>
    <Image
      source={
        Object {
          "uri": "http://ddragon.leagueoflegends.com/cdn/7.2.1/img/profileicon/553.png",
        }
      }
      style={
        Object {
          "borderRadius": 35,
          "height": 70,
          "width": 70,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "dodgerblue",
          "fontSize": 16,
          "padding": 10,
          "textAlign": "center",
        }
      }>
       
      Test
       
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
            "fontWeight": "bold",
          },
          Object {},
          Array [
            Object {
              "color": "dodgerblue",
            },
          ],
        ]
      }>
      Division Name
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
          Object {},
        ]
      }>
      Trundle\'s Dervish
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
            "fontWeight": "bold",
          },
          Object {},
          Array [
            Object {
              "color": "dodgerblue",
            },
          ],
        ]
      }>
      Tier
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
          Object {},
        ]
      }>
      SILVER
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
            "fontWeight": "bold",
          },
          Object {},
          Array [
            Object {
              "color": "dodgerblue",
            },
          ],
        ]
      }>
      Wins
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
          Object {},
        ]
      }>
      21
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
            "fontWeight": "bold",
          },
          Object {},
          Array [
            Object {
              "color": "dodgerblue",
            },
          ],
        ]
      }>
      Losses
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
          Object {},
        ]
      }>
      13
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
            "fontWeight": "bold",
          },
          Object {},
          Array [
            Object {
              "color": "dodgerblue",
            },
          ],
        ]
      }>
      LP
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
          Object {},
        ]
      }>
      26
    </Text>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "dodgerblue",
            },
          ]
        }>
        Recent Games
      </Text>
    </View>
  </View>
</View>
`;
